packages/carbon/src/plugins/linked-roles/LinkedRoles.ts

```diff
@@ -48,64 +49,76 @@ type Tokens = {
  * })
  * ```
  */
-export class LinkedRoles {
+export class LinkedRoles extends Plugin {
 	client: Client
 	options: Required<LinkedRolesOptions>
 
+	// TODO: I would like to remove the need to pass the client here
```

Comment:
TODO reminder (?)

My first thought on this is to refactor the constructor to be something in the shape of
```
constructor(options: LinkedRolesOptions, data: {client?: Client, env: Env})
```
but I haven't looked into how its laid out

Reply:
I was thinking more of just passing the options that linked roles grabs from the client directly into the plugin itself. The env would sort of be used more than once but I prefer that over dependency injection in this case.

I had another idea of an `Env` class, that might be used like such:
```ts
const env = new Env({ clientId: env.DISCORD_CLIENT_ID })
const client = new Client({ env, ... })
const linkedRoles = new LinkedRoles({ env, ... })
return [client, linkedRoles]
```
Though I'm not the biggest fan of this.

Reply:
This might be part of another pr